extern/CMakeProject-mad.cmake

```diff
@@ -38,6 +49,35 @@ disable_project_warnings("mad")
 
 if(MSVC)
   sm_add_compile_definition("mad" _CRT_SECURE_NO_WARNINGS)
+  sm_add_compile_definition("mad" ASO_ZEROCHECK)
+  sm_add_compile_definition("mad" $<$<CONFIG:Debug>:FPM_DEFAULT>)
+  sm_add_compile_definition("mad" $<$<CONFIG:Release>:FPM_INTEL>)
+  sm_add_compile_definition("mad" $<$<CONFIG:MinSizeRel>:FPM_INTEL>)
+  sm_add_compile_definition("mad" $<$<CONFIG:RelWithDebInfo>:FPM_INTEL>)
+  sm_add_compile_definition("mad" HAVE_ASSERT_H=1)
+  sm_add_compile_definition("mad" HAVE_ERRNO_H=1)
+  sm_add_compile_definition("mad" HAVE_FCNTL_H=1)
+  sm_add_compile_definition("mad" HAVE_INTTYPES_H=1)
+  sm_add_compile_definition("mad" HAVE_LIMITS_H=1)
+  sm_add_compile_definition("mad" HAVE_MEMORY_H=1)
+  sm_add_compile_definition("mad" HAVE_STDINT_H=1)
+  sm_add_compile_definition("mad" HAVE_STDLIB_H=1)
+  sm_add_compile_definition("mad" HAVE_STRINGS_H=1)
+  sm_add_compile_definition("mad" HAVE_STRING_H=1)
+  sm_add_compile_definition("mad" HAVE_SYS_STAT_H=1)
+  sm_add_compile_definition("mad" HAVE_SYS_TYPES_H=1)
```

Comment:
I'll need to see if there is a way to have most of these determined and defined within cmake itself. Right now it seems like we are assuming.